File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
def solution(arr):
    answer = []
    n = len(arr)
    
    answer.append(arr[0])
    
    if(n <= 1):
        return answer
    
    for i in range(1, n):
        if arr[i] != answer[-1]:
            answer.append(arr[i])
    
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    int count = 0;
    answer.push_back(arr[0]);
    
    for(int i = 1; i < arr.size(); ++i){
        if(answer[count] != arr[i]){
            answer.push_back(arr[i]);
            count++;
       }
    }
    
    return answer;
}